run/logging-manual/src/main/java/com/example/cloudrun/App.java

```diff
@@ -42,11 +42,12 @@ public static void main(String[] args) {
         (req, res) -> {
           // [START cloudrun_manual_logging]
           // [START run_manual_logging]
-          // [START functions_manual_logging]
           // Build structured log messages as an object.
           Object globalLogFields = null;
-          // [END functions_manual_logging]
+
           // Add log correlation to nest all log messages beneath request log in Log Viewer.
+          // TODO(developer): delete this code if you're creating a Cloud
+          //                  Function and it is *NOT* triggered by HTTP.
           String traceHeader = req.headers("x-cloud-trace-context");
           if (traceHeader != null && project != null) {
```

Comment:
Does `traceHeader != null` not prevent the code to be deleted from executing?

Reply:
It does, but if `req` is not defined I'm thinking this would throw a compiler error.

Reply:
`req` _is_ declared, on line 42. The issue is that _`get`_ is not declared, so there's no way to know what the `(req, res) -> {...}` lambda is supposed to mean.